File: src/main/java/com/kingmang/ixion/ast/ReturnStatement.java
package com.kingmang.ixion.ast;

import com.kingmang.ixion.StatementVisitor;
import com.kingmang.ixion.lexer.Position;

public final class ReturnStatement extends Statement {
    public final Expression expression;

    public ReturnStatement(Position pos, Expression expression) {
        super(pos);
        this.expression = expression;
    }

    public <R> R accept(StatementVisitor<R> visitor) {
        return visitor.visitReturnStmt(this);
    }
}

File: src/main/java/com/kingmang/ixion/ast/ReturnStatement.kt
package com.kingmang.ixion.ast

import com.kingmang.ixion.StatementVisitor
import com.kingmang.ixion.lexer.Position

class ReturnStatement(pos: Position?, @JvmField val expression: Expression?) : Statement(pos) {

    override fun <R> accept(visitor: StatementVisitor<R?>?): R? {
        return visitor?.visitReturnStmt(this)
    }
}